cypress/page/student-registration-form.page.ts

```diff
@@ -0,0 +1,75 @@
+class StudentRegistrationForm {
+  private nameText: string;
+  private lastNameText: string;
+  private emailText: string;
+  private genderBtn: string;
+  private mobileNumberText: string;
+  private hobbiesCheck: string;
+  private currentAddressText: string;
+  private submitBtn: string;
+  private stateList: string;
+  private cityList: string;
+  private NCROstate: string;
+  private delhiCity: string;
+
+  constructor() {
+    this.studentFormPageURL = 'https://demoqa.com/automation-practice-form';
+    this.bodyTr = 'tbody tr';
+    this.nameText = '#firstName'
+    this.lastNameText = '#lastName';
+    this.emailText = '#userEmail';
+    this.genderBtn = '[type="radio"]';
+    this.mobileNumberText = '#userNumber';
+    this.hobbiesCheck = '[type="checkbox"]';
+    this.currentAddressText = '#currentAddress';
+    this.submitBtn = '#submit';
+    this.stateList = '#state > div > div.css-1hwfws3 > div.css-1wa3eu0-placeholder';
+    this.cityList = '#city > div > div.css-1hwfws3 > div.css-1wa3eu0-placeholder';
+    this.NCROstate = "#react-select-3-option-0";
+    this.delhiCity = "#react-select-4-option-0";
+  }
+
+  public visitStudentForm(): void {
+    cy.visit(this.studentFormPageURL);
+  }
+
+  fillForm(personalInformation: {
+      name: string, 
+      lastName: string, 
+      email: string,
+      mobileNumber: string, 
+      hobbies: string[], 
+      currentAddress: string,
+      state: string,
+      city: string
+  }) {
+      cy.get(this.nameText).type(personalInformation.name);
+      cy.get(this.lastNameText).type(personalInformation.lastName);
+      cy.get(this.emailText).type(personalInformation.email);
+      cy.get(this.genderBtn).check("Male", {force: true});
+      cy.get(this.mobileNumberText).type(personalInformation.mobileNumber);
+      cy.get(this.hobbiesCheck).check(personalInformation.hobbies, {force: true});
```

Comment:
you can define an internal function or object to match strings to number values here

Reply:
resolved